engines/ml/lightgbm/src/main/java/ai/djl/ml/lightgbm/jni/JniUtils.java

```diff
@@ -92,12 +100,29 @@ public static double[] inferenceMat(SWIGTYPE_p_p_void model, int iterations, Lgb
                             outLength,
                             outBuffer);
             checkCall(result);
-            long length = lightgbmlib.int64_tp_value(outLength);
-            double[] values = new double[(int) length];
-            for (int i = 0; i < length; i++) {
-                values[i] = lightgbmlib.doubleArray_getitem(outBuffer, i);
+            int length = Math.toIntExact(lightgbmlib.int64_tp_value(outLength));
+            if (a.getDataType() == DataType.FLOAT32) {
+                ByteBuffer bb = ByteBuffer.allocateDirect(length * 4);
+                FloatBuffer wrapped = bb.asFloatBuffer();
+                for (int i = 0; i < length; i++) {
+                    wrapped.put((float) lightgbmlib.doubleArray_getitem(outBuffer, i));
```

Comment:
Can we call `lightgbm.floatArray_getitem()`?

Reply:
The behavior of the lightgbm engine is to always return a double array and this is the utility provided by lightgbm to get from a JNI `double[]` data structure. So if we want to return a float[], we need to convert it from double to float